src/algo.rs

```diff
@@ -10,6 +10,25 @@ pub struct Profile {
     pub arr_time: u16,
 }
 
+impl Profile {
+    pub fn route(&self, profiles: &Vec<Vec<Profile>>, timetable: &Timetable) -> Vec<Connection> {
```

Comment:
by curiosity why do you return `Connection` and not a `&`

since they belong to `timetable` it seems to me the `timetable` can outlive the return value (but it sure is more convenient and it will be easier to change the `timetable` for updates if needed)

Reply:
I’m a bit confused what exactly I want as a parameter. A slice of Vec ? `&[Vec<Profile>]` ? I end up having `&&` in the function calls.

For `Connection`, I’m still not comfortable with references ;)

Reply:
Are you happy now? my first lifetime annotation ;)